### PR TITLE
Avoids duplicate lectures

### PR DESCRIPTION
**Pull request by jaysaleh**
This pull request avoid duplicate lecture discussion from being created.

### CODE REVIEW

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 1]
```diff
@@ -66,6 +67,13 @@ public void init() throws ServletException {
   // TODO: Check and see if lectureURL is already in database and if it is valid.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Entity lectureEntity = createLectureEntity(request);
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:69]: Can this be moved to after the isPresent() check? Only create it if it is going to be added

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 2]
```diff
@@ -66,6 +67,13 @@ public void init() throws ServletException {
   // TODO: Check and see if lectureURL is already in database and if it is valid.
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:67]: Remove the first part of this TODO?

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 3]
```diff
@@ -78,6 +86,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(gson.toJson(lectures));
   }
 
+  /**
+   * Returns the Entity in database that has {@code url}, or an
+   * empty optional if non exist.
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:91]: Can you use `{@code Optional.empty()}`? Also `if one doesn't exist.`

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 4]
```diff
@@ -66,6 +67,13 @@ public void init() throws ServletException {
   // TODO: Check and see if lectureURL is already in database and if it is valid.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:68]: Can you include @Override?

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 3 above)
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:91]: if none exist?

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 1 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:69]: fixed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 2 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:67]: fixed

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 4 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:68]: @override is included right above the todo

Comment (amuamushu) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:68]: Oops sorry I must have missed it.

**Pull request merged**